src/buf_writer.c

```diff
@@ -134,3 +134,33 @@ buf_writer_terminate(tsdn_t *tsdn, buf_writer_t *buf_writer) {
 		buf_writer_free_internal_buf(tsdn, buf_writer->buf);
 	}
 }
+
+void
+buf_writer_set_allow_auto_flush(buf_writer_t *buf_writer,
+    bool allow_auto_flush) {
+	/* TODO: set allow_auto_flush */
+}
+
+void
+buf_writer_commit(buf_writer_t *buf_writer) {
+	/* TODO: commit */
+}
+
+bool
+buf_writer_overflow(buf_writer_t *buf_writer) {
+	/* TODO: check overflow */
+	return false;
+}
+
+void
+buf_writer_revert(buf_writer_t *buf_writer) {
```

Comment:
Question: should we make the revert implicit? meaning if it's overflowed, revert the uncommitted stuff automatically.

If we keep the revert call, maybe name it  `_revert_uncommitted`? Also the _overflow above might be better as _overflowed.

Reply:
Good point. Will change. Will also implement the internals in `buf_writer` since the outer logic has pretty much been settled.